hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieMergeOnReadRDD.scala

```diff
@@ -116,7 +125,45 @@ class HoodieMergeOnReadRDD(@transient sc: SparkContext,
       Option(TaskContext.get()).foreach(_.addTaskCompletionListener[Unit](_ => iter.asInstanceOf[Closeable].close()))
     }
 
-    iter
+    val commitTimeMetadataFieldIdx = requiredSchema.structTypeSchema.fieldNames.indexOf(HoodieRecord.COMMIT_TIME_METADATA_FIELD)
+    val needsFiltering = commitTimeMetadataFieldIdx >= 0 && !StringUtils.isNullOrEmpty(startTimestamp) && !StringUtils.isNullOrEmpty(endTimestamp)
```

Comment:
Does the `requiredSchema` always include `_hoodie_commit_time` meta field? When this field could missing, we need also hanlde the case to keep correctness.

Reply:
The `_hoodie_commit_time` field is now mandatory only for `MergeOnReadIncrementalRelation`.
Queries of other table types do not use the parquet record filter function.

https://github.com/apache/hudi/blob/9d58ee4b1f1fce213ee3f4ff478eb003da943924/hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/MergeOnReadIncrementalRelation.scala#L200-L205

Reply:
So, for the use case we wanna fix, this meta field always exists.